src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/MonitoringSourceConfiguration.cs

```diff
@@ -28,12 +28,55 @@ public abstract class MonitoringSourceConfiguration
         public const string EventPipeProviderName = "Microsoft-DotNETCore-EventPipe";
         public const string SystemDiagnosticsMetricsProviderName = "System.Diagnostics.Metrics";
 
+        private bool _requestRundown = true;
+        private long _rundownKeyword = EventPipeSession.DefaultRundownKeyword;
+
         public static IEnumerable<string> DefaultMetricProviders => new[] { SystemRuntimeEventSourceName, MicrosoftAspNetCoreHostingEventSourceName, GrpcAspNetCoreServer };
 
         public abstract IList<EventPipeProvider> GetProviders();
 
-        public virtual bool RequestRundown { get; set; } = true;
+        public virtual bool RequestRundown
+        {
+            get => _requestRundown;
+            set
+            {
+                _requestRundown = value;
+                if (_requestRundown)
+                {
+                    if (_rundownKeyword == 0)
+                    {
+                        _rundownKeyword = EventPipeSession.DefaultRundownKeyword;
+                    }
+                }
+                else
+                {
+                    _rundownKeyword = 0;
+                    RetryStrategy = RetryStrategy.DoNotRetry;
```

Comment:
I don't think we should automatically change the retry policy, instead we should expect the owner of this object to set the policy appropriately for their scenario. We can set the defaults as RundownKeyword = DefaultRundownKeyword and Retry = DropKeywordKeepRundown. Any place the code currently creates a config with RequestRundown = false would need to be updated but GC profile is probably the only one that does.